src/settings.ts

```diff
@@ -136,6 +136,15 @@ export namespace Settings {
         label: i18n.__("settings.createSubfolder.label"),
         description: i18n.__("settings.createSubfolder.description"),
       },
+      createSubfolderPerProfile: {
+        value: false,
+        type: SettingItemType.Bool,
+        section: "backupSection",
+        public: true,
+        advanced: true,
+        label: i18n.__("settings.createSubfolderPerProfile.label"),
+        description: i18n.__("settings.createSubfolderPerProfile.description"),
```

Comment:
It might make sense to use `SettingStorage.File` here so that this setting can be backed up. This wouldn't be consistent with the other settings though.

Reply:
My opinion is that there must be a Joplin API to backup all plugin settings and restore settings from selected plugins. 
In my opinion, SettingStorage.File is not a good idea and not a good way to proceed. If so, Joplin should automatically convert all settings in one version to SettingStorage.File and not provide x ways. 
I also don't understand why all the settings from the database have to be in a json file. But that is a different topic.